libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -906,20 +906,16 @@ - (void)clearAccountState:(BOOL)clearAccountData {
         });
         return;
     }
-    
     if (clearAccountData) {
         [SFSecurityLockout clearPasscodeState];
+        self.idCoordinator.idData = nil;
```

Comment:
The `YES` portion of `clearAccountState:` is meant to cover scenarios where the user is logging out.  These lines, OTOH, are meant to clear the local state of the authentication components, prior to (presumably) initiating a new login process.

I don't think these belong here.